src/backend/bacon-video-widget-gst-missing-plugins.c

```diff
@@ -306,196 +306,78 @@ bacon_video_widget_start_plugin_installation (TotemCodecInstallContext *ctx,
-		g_warning ("Couldn't find codec related information");
-		return NULL;
-	}
-
-	ss = g_strsplit (split[4], "-", 2);
-	type_name = g_strdup (ss[0]);
-	caps = g_strdup (ss[1]);
-
-	s = gst_structure_from_string (caps, NULL);
-	if (s == NULL) {
-		g_warning ("totem: failed to parse caps: %s", caps);
-		return NULL;
-	}
-
-	info = g_new0 (CodecInfo, 1);
-	info->type_name = g_steal_pointer (&type_name);
-	info->structure = s;
-	return info;
-}
-
 static void
-codec_info_free (CodecInfo *codec)
+determine_action_for_missing_codecs (TotemCodecInstallContext	 *ctx,
+				     const gchar		 *install_helper_display_name,
+				     gchar			**button_text,
+				     gchar			**message_text)
 {
-	gst_structure_free (codec->structure);
-	g_free (codec->type_name);
-	g_free (codec);
-}
-
-static gboolean
-missing_codecs_supported (const char **codecs)
-{
-	g_autoptr(GPtrArray) codecs_array = NULL;
-	guint len, i;
-
-	/* First of all filter the detailed strings for each missing codec
-	 * so that we only consider those we're able to successfully parse. */
-	len = g_strv_length ((GStrv)codecs);
-	codecs_array = g_ptr_array_new_with_free_func ((GDestroyNotify) codec_info_free);
-	for (i = 0; i < len; i++) {
-		CodecInfo *info = codec_info_new (codecs[i]);
-		if (info == NULL)
-			continue;
-
-		g_ptr_array_add (codecs_array, info);
+	GDesktopAppInfo *candidate;
+
+	if (install_helper_display_name != NULL) {
+		gchar *descriptions_text;
+		descriptions_text = g_strjoinv (", ", ctx->descriptions);
+		/* TRANSLATORS: this is a button to launch a codec installer.
+		 * %s will be replaced with the software installer's name, e.g.
+		 * 'Software' in case of gnome-software. */
+		*button_text = g_strdup_printf (_("_Find in %s"), install_helper_display_name);
+		*message_text = g_strdup_printf (ngettext ("%s is required to play the file, but is not installed.",
+							   "%s are required to play the file, but are not installed.",
+							   g_strv_length (ctx->descriptions)),
+						 descriptions_text);
+		g_free (descriptions_text);
+		return;
 	}
 
-	/* No successfully parsed codecs strings, we can't tell */
-	if (codecs_array->len == 0)
-		return FALSE;
-
-	/* All the codecs must be supported for us to be able to declare that
-	 * we support the missing codecs with our codecs pack */
-	for (i = 0; i < codecs_array->len; i++) {
-		CodecInfo *info = g_ptr_array_index (codecs_array, i);
-
-		if (info->structure == NULL)
-			return FALSE;
-
-		/* No encoders are supported */
-		if (g_strcmp0 (info->type_name, "decoder") != 0)
-			return FALSE;
-
-		if (!is_supported_codec (info->structure))
-			return FALSE;
+	/* First try org.videolan.VLC.desktop, and if it exists
+	 * use that, otherwise use eos-vlc.desktop */
+	candidate = g_desktop_app_info_new ("org.videolan.VLC.desktop");
+	if (candidate) {
+		*message_text = g_strdup (_("This file type is currently unsupported in Videos. "
+					    "video player. However, you can use VLC media player "
```

Comment:
One of our paid translators noticed an error in this source string: "Videos. video player."  I assume the "video player." should be removed, so I'll clean it up.

Reply:
Outch, I remember we had some changes to the wording during the review, odd that it slipped in. Thanks for fixing Roddy.

Reply:
There were two very similar strings.  Perhaps we fixed the other one and missed this one.  No worries.